packages/babel-plugin/src/css-prop/__tests__/behaviour.test.tsx

```diff
@@ -390,4 +390,101 @@ describe('css prop behaviour', () => {
 
     expect(actual).toInclude(':hover{color:red}');
   });
+
+  it('should apply conditional logical expression object spread styles', () => {
+    const actual = transform(`
+      import '@compiled/react';
+      import React from 'react';
+
+      const Component = props => (
+        <div css={{
+          ...props.isPrimary && {
+            color: 'blue',
+            fontSize: 20,
+          }
+        }}>hello world</div>
+      );
+    `);
+
+    expect(actual).toInclude('className={ax([props.isPrimary&&"_syaz13q2 _1wybgktf"])}');
```

Comment:
This is the place where compilation can't help much because `isPrimary` won't be known until runtime. Love to compact this `[props.isPrimary&&"_syaz13q2 _1wybgktf"]` into `[]` if `props.isPrimary` evaluates to `false`. But its not possible. Though bundlers can do some magic if `isPrimary` is statically defined like `<Component  isPrimary />`. Not sure.

Reply:
yeah thats a head scratcher - we could always put in extra code to see if any code resolves to false but it's questionable if it's worth the effort.

maybe folder for later?

Reply:
yeah later only. I was just discussing whether this is possible or not.